src/Microsoft.Health.Fhir.Shared.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -529,7 +529,7 @@ private IEnumerable<IncludeExpression> ParseIncludeIterateExpressions(IList<(str
                     allowedResourceTypesByScope = _contextAccessor.RequestContext?.AccessControlContext?.AllowedResourceActions.Select(s => s.Resource);
                 }
 
-                var expression = _expressionParser.ParseInclude(includeResourceTypeList, p.query, isReversed, iterate, allowedResourceTypesByScope);
+                var expression = _expressionParser.ParseInclude(includeResourceTypeList, p.query, isReversed, iterate, allowedResourceTypesByScope.ToList());
```

Comment:
ToList can't be used here without a null check. This is probably causing some of the test failures.